process/block/preprocess/transactions.go

```diff
@@ -212,6 +244,38 @@ func (txs *transactions) ProcessBlockTransactions(body block.Body, round uint64,
 			continue
 		}
 
+		mapHashTx, err := txs.getMapHashTxFromMiniBlock(miniBlock)
+		if err != nil {
+			return err
+		}
+
+		gasConsumedByMiniBlockInSenderShard, gasConsumedByMiniBlockInReceiverShard, err := txs.gasHandler.ComputeGasConsumedByMiniBlock(
+			miniBlock,
+			mapHashTx)
+
+		if err != nil {
+			return err
+		}
+
+		var gasConsumedInSelfShard uint64
+		if txs.shardCoordinator.SelfId() == miniBlock.SenderShardID {
+			gasConsumedInSelfShard = gasConsumedByMiniBlockInSenderShard
+
+			if gasConsumedByMiniBlockInReceiverShard > txs.economicsFee.MaxGasLimitPerBlock() {
+				return process.ErrMaxGasLimitPerMiniBlockInReceiverShardIsReached
+			}
+		} else {
+			gasConsumedInSelfShard = gasConsumedByMiniBlockInReceiverShard
+
+			if gasConsumedByMiniBlockInSenderShard > txs.economicsFee.MaxGasLimitPerBlock() {
+				return process.ErrMaxGasLimitPerMiniBlockInSenderShardIsReached
```

Comment:
you are really obliged to process this. even if you do not like it. It would block your shard completely to process cross shard miniblocks.

Reply:
If this case appears that means that this block is created by a proposer which didn't take into account the consumed gas limitation and you should not sign this block. This method is called only on ProcessBlock